.github/workflows/production-heartbeat.yml

```diff
@@ -11,28 +11,148 @@ on:
+          if code --list-extensions | grep -q 'salesforce.sfdx-code-analyzer-vscode'; then
+            echo "Extension installed successfully"
+          else
+            echo "Extension installation failed" && exit 1
+          fi
+
+      # 2 Install VS Code and Extension on Windows
+      # We use chocolatey to install vscode since it gives a reliable path for the location of code.exe
+      # We have also seen Windows to be flaky, so adding addition echo statements.
+      - name: Install VS Code on Windows
+        if: runner.os == 'Windows'
+        run: |
+          Write-Host "Installing Chocolatey..."
+          Set-ExecutionPolicy Bypass -Scope Process -Force;
+          [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072;
+          iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+          Write-Host "Installing Visual Studio Code using Chocolatey..."
+          choco install vscode -y
+
+      - name: Install Salesforce Code Analyzer Extension on Windows
+        if: runner.os == 'Windows'
+        run: |
+          echo "Installing Code Analyzer Extension..."
+          "/c/Program Files/Microsoft VS Code/bin/code" --install-extension salesforce.sfdx-code-analyzer-vscode
+          echo "Installing Code Analyzer Complete"
+          
+          echo "Waiting for 10 seconds..."
+          sleep 10
+          
+          echo "Listing installed extensions..."
+          "/c/Program Files/Microsoft VS Code/bin/code" --list-extensions
+        shell: bash
+
+      - name: Verify Extension on Windows
+        if: runner.os == 'Windows'
+        run: |
+          echo "Waiting for 10 seconds..."
+          sleep 10
+
+          echo "Listing installed extensions..."
+          extensions=$("/c/Program Files/Microsoft VS Code/bin/code" --list-extensions)
+
+          echo "$extensions"
+
+          if echo "$extensions" | grep -q 'salesforce.sfdx-code-analyzer-vscode'; then
+            echo "Extension 'salesforce.sfdx-code-analyzer-vscode' is installed successfully"
+          else
+            echo "Extension 'salesforce.sfdx-code-analyzer-vscode' is NOT installed"
+            exit 1
+          fi
+        shell: bash
+
+      # 3 Install VS Code and Extension on macOS
+      - name: Install VS Code on macOS
+        if: runner.os == 'macOS'
+        run: |
+          brew install --cask visual-studio-code
+
+      - name: Install Salesforce Code Analyzer Extension on macOS
+        if: runner.os == 'macOS'
+        run: |
+          code --install-extension salesforce.sfdx-code-analyzer-vscode
+
+      - name: Verify Extension Installation on macOS
+        if: runner.os == 'macOS'
+        run: |
+          if code --list-extensions | grep -q 'salesforce.sfdx-code-analyzer-vscode'; then
+            echo "Extension installed successfully"
+          else
+            echo "Extension installation failed" && exit 1
+          fi
```

Comment:
These steps seem to be the same as ubuntu. Any reason why we didn't reuse this code?